### PR TITLE
streamlined Heroku buildpack

### PR DESCRIPTION
**Pull request by abernicchia-heroku**
- removed env-to-config that had limited config options and couldn't handle Heroku DATABASE_URL
- no need for postgrest.conf file in favour of Heroku environment variables as per https://12factor.net/config
- no hardcoded Procfile that shouldn't be created by the buildpack, now it's more flexible and can be modified if needed. By default, it handles both the Heroku DATABASE_URL and external db uri via PGRST_DB_URI
- removed unused code from bin/compile
- new step-by-step How To (README.md) with test cases
- renamed racially loaded terms such as whitelist_regex / blacklist_regex, now allowlist_regex / blocklist_regex
- POSTGREST_VER default is now 10.0.0 (previously 9.0.1)
- Supported versions only >= 9 - this removes most of the cumbersome code due to old releases. Only new releases are probably expected to be used with the new/forthcoming PostgreSQL releases (PG14/15...)

### CODE REVIEW

**Review by steve-chavez — APPROVED** [review]
> Supported versions only >= 9 - this removes most of the cumbersome code due to old releases. Only new releases are probably expected to be used with the new/forthcoming PostgreSQL releases (PG14/15...)

Correct. The change LGTM!

**Comment by abernicchia-heroku** [comment]
Now it's also necessary to publish the buildpack so the Heroku Elements page (basically README.md) for the buildpack (https://elements.heroku.com/buildpacks/postgrest/postgrest-heroku) is aligned with GitHub too - see https://devcenter.heroku.com/articles/buildpack-registry#publishing-a-buildpack-version

File: README.md [hunk 1]
```diff
@@ -30,56 +22,96 @@ Alternatively you could create it on [ElephantSQL](https://www.elephantsql.com/)
 2.  Create a new Heroku app using this Heroku buildpack:
 
     ```bash
-    heroku apps:create ${YOUR_APP_NAME} --buildpack https://github.com/PostgREST/postgrest-heroku
+    mkdir ${YOUR_APP_NAME}
+    cd ${YOUR_APP_NAME}
+    git init .
+
+    heroku apps:create ${YOUR_APP_NAME} --buildpack https://github.com/abernicchia-heroku/postgrest-heroku.git
```
Comment (steve-chavez) [inline README.md:29]: > https://github.com/abernicchia-heroku/postgrest-heroku.git
 
Oops, the above URL should be changed to https://github.com/PostgREST/postgrest-heroku.git right?

Comment (abernicchia-heroku) [reply README.md:29]: yes, exactly!

**Pull request merged**